hapi/src/api/offer.api.js

```diff
@@ -0,0 +1,37 @@
+const accountApi = require('./account.api')
+const userApi = require('./user.api')
+
+const redeem = async (from, details) => {
+  const user = await userApi.getOne({
+    account: { _eq: from }
+  })
+
+  if (user.role !== 'donor') {
+    throw new Error('Only donors can redeem an offer')
+  }
+
+  const userTo = await userApi.getOne({
+    account: { _eq: details.to }
+  })
+
+  if (userTo.role !== 'sponsor') {
+    throw new Error('Only sponsor can receive tokens by an offer redemption')
```

Comment:
These errors are presented in a snackbar component?

Reply:
It is just Backend, front-end integration is in another card

Reply:
Got it